opensafely/notebook.py

```diff
@@ -0,0 +1,71 @@
+import json
+from pathlib import Path
+import os
+import subprocess
+import sys
+import time
+import webbrowser
+
+
+DESCRIPTION = "Run a jupyter notebook using the OpenSAFELY environment"
+
+def add_arguments(parser):
+    pass
+
+
+def docker(*args, **kwargs):
+    cmd = ("docker",) + args
+    check = kwargs.get("check", True)
+    capture = kwargs.get("capture_output", True)
+    newlines = kwargs.get("universal_newlines", True)
+    return subprocess.run(cmd, check=check, capture_output=capture, universal_newlines=newlines)
+
+
+def main():
+    container = None
+    pwd = Path(os.getcwd())
+    name = f"opensafely-notebook-{pwd.name}"
+    launch = [
+            "run",
+            # we initially detach, so we can find out the ip address.
+            "--detach",
+            "--rm",
+            "--init",
+            # we force tty now so we can re-attach stdin/out later
+            "-it",
+            f"--name={name}",
+            f"--hostname={name}",
+            "--label=opensafely",
+            "-v",
+            f"/{pwd}:/workspace",
+            "ghcr.io/opensafely-core/python",
+            "jupyter",
+            "lab",
+            "--ip",
+            "0.0.0.0",
+            # python image currently runs as root
+            "--allow-root",
+            # disable authentication, as it's only accessible via direct access
+            # to the docker container
+            "--LabApp.token=''",
+        ]
+
+    launch_ps = docker(*launch)
+    container = launch_ps.stdout.strip()
```

Comment:
I don't actually think you need `container` because you can pass `name` to `docker inspect` and it will work exactly the same.

Reply:
oh! yes, brilliant. I only added name at the end.